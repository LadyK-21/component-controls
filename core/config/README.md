# Table of contents

-   [Overview](#overview)
-   [Installation](#installation)
-   [API](#api)
    -   [ConfigurationResult](#configurationresult)
    -   [RequireContextProps](#requirecontextprops)
    -   [buildConfigFileNames](#buildconfigfilenames)
    -   [optionsFileNames](#optionsfilenames)
    -   [configRequireContext](#configrequirecontext)
    -   [extractDocuments](#extractdocuments)
    -   [fixGlob](#fixglob)
    -   [getConfigurationArg](#getconfigurationarg)
    -   [loadConfiguration](#loadconfiguration)
    -   [mergeBuildConfiguration](#mergebuildconfiguration)

# Overview

Configuration file utilities. Uses the [glob](https://www.npmjs.com/package/glob) package to locate story files from the configuration

# Installation

This package is usually installed as part of the @component-controls package, but you can also install it standalone:

```bash
$ npm install @component-controls/config --save-dev
```

# API

<tsdoc-typescript entry="./src/index.ts" files="../specification/src/configuration.ts"/>

<!-- START-TSDOC-TYPESCRIPT -->

## ConfigurationResult

_defined in [@component-controls/config/src/index.ts](https://github.com/ccontrols/component-controls/tree/master/core/config/src/index.ts#L26)_



### properties

| Name              | Type                                      | Description |
| ----------------- | ----------------------------------------- | ----------- |
| `config*`         | [BuildConfiguration](#buildconfiguration) |             |
| `configPath*`     | string                                    |             |
| `optionsFilePath` | string                                    |             |

## RequireContextProps

from the glob list of documents, extract require.context array of props

_defined in [@component-controls/config/src/index.ts](https://github.com/ccontrols/component-controls/tree/master/core/config/src/index.ts#L130)_



### properties

| Name                 | Type              | Description |
| -------------------- | ----------------- | ----------- |
| `directory*`         | string            |             |
| `regExp*`            | [RegExp](#regexp) |             |
| `useSubdirectories*` | boolean           |             |

## buildConfigFileNames

_defined in [@component-controls/config/src/index.ts](https://github.com/ccontrols/component-controls/tree/master/core/config/src/index.ts#L11)_



## optionsFileNames

_defined in [@component-controls/config/src/index.ts](https://github.com/ccontrols/component-controls/tree/master/core/config/src/index.ts#L18)_



## configRequireContext

convert glob patters from config file into require.context objects

_defined in [@component-controls/config/src/index.ts](https://github.com/ccontrols/component-controls/tree/master/core/config/src/index.ts#L138)_

**function** configRequireContext(`__namedParameters`\*: **config**: ### properties| Name         | Type                                | Description                                                                                                                                    |
| ------------ | ----------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------- |
| `BuildProps` | [BuildProps](#buildprops)           |                                                                                                                                                |
| `categories` | [DocType](#doctype)\[]              | page types that are considered as categories fields as well                                                                                    |
| `files`      | string \| string\[]                 | alternative naming for docz compatibility                                                                                                      |
| `ignore`     | string\[]                           | files to ignore. by default \['readme.md', 'changelog.md', 'code_of_conduct.md', 'contributing.md', 'license.md']                              |
| `instrument` | any                                 | instrumentation configuration                                                                                                                  |
| `pages`      | [PagesOnlyRoutes](#pagesonlyroutes) | base url path for API documentation pages. Default is "docs/"                                                                                  |
| `siteMap`    | [SitemapConfig](#sitemapconfig)     | if false, disable automatic sitemap generation                                                                                                 |
| `siteRoot`   | string                              | the site base url, by default the site starts at /                                                                                             |
| `siteUrl`    | string                              | Deployed site url. Also used for auto generated sitemap.                                                                                       |
| `stories`    | string \| string\[]                 | wild card search string for the stories internally using \`glob\` for the search: https&#x3A;//www.npmjs.com/package/glob example: "./stories/ |**configPath**: string): [RequireContextProps](#requirecontextprops)\[] | undefined;

### parameters

| Name                 | Type                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            | Description |
| -------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------- |
| `__namedParameters*` | **config**: ### properties| Name         | Type                                | Description                                                                                                                                    | | ------------ | ----------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------- | | `BuildProps` | [BuildProps](#buildprops)           |                                                                                                                                                | | `categories` | [DocType](#doctype)\[]              | page types that are considered as categories fields as well                                                                                    | | `files`      | string \| string\[]                 | alternative naming for docz compatibility                                                                                                      | | `ignore`     | string\[]                           | files to ignore. by default \['readme.md', 'changelog.md', 'code_of_conduct.md', 'contributing.md', 'license.md']                              | | `instrument` | any                                 | instrumentation configuration                                                                                                                  | | `pages`      | [PagesOnlyRoutes](#pagesonlyroutes) | base url path for API documentation pages. Default is "docs/"                                                                                  | | `siteMap`    | [SitemapConfig](#sitemapconfig)     | if false, disable automatic sitemap generation                                                                                                 | | `siteRoot`   | string                              | the site base url, by default the site starts at /                                                                                             | | `siteUrl`    | string                              | Deployed site url. Also used for auto generated sitemap.                                                                                       | | `stories`    | string \| string\[]                 | wild card search string for the stories internally using \`glob\` for the search: https&#x3A;//www.npmjs.com/package/glob example: "./stories/ |**configPath**: string |             |
| `returns`            | [RequireContextProps](#requirecontextprops)\[] \| undefined                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     |             |

## extractDocuments

find the story files out of a configuration file
using glob for the regex file search

_defined in [@component-controls/config/src/index.ts](https://github.com/ccontrols/component-controls/tree/master/core/config/src/index.ts#L104)_

**function** extractDocuments(`__namedParameters`\*: **config**: ### properties| Name         | Type                                | Description                                                                                                                                    |
| ------------ | ----------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------- |
| `BuildProps` | [BuildProps](#buildprops)           |                                                                                                                                                |
| `categories` | [DocType](#doctype)\[]              | page types that are considered as categories fields as well                                                                                    |
| `files`      | string \| string\[]                 | alternative naming for docz compatibility                                                                                                      |
| `ignore`     | string\[]                           | files to ignore. by default \['readme.md', 'changelog.md', 'code_of_conduct.md', 'contributing.md', 'license.md']                              |
| `instrument` | any                                 | instrumentation configuration                                                                                                                  |
| `pages`      | [PagesOnlyRoutes](#pagesonlyroutes) | base url path for API documentation pages. Default is "docs/"                                                                                  |
| `siteMap`    | [SitemapConfig](#sitemapconfig)     | if false, disable automatic sitemap generation                                                                                                 |
| `siteRoot`   | string                              | the site base url, by default the site starts at /                                                                                             |
| `siteUrl`    | string                              | Deployed site url. Also used for auto generated sitemap.                                                                                       |
| `stories`    | string \| string\[]                 | wild card search string for the stories internally using \`glob\` for the search: https&#x3A;//www.npmjs.com/package/glob example: "./stories/ |**configPath**: string): string\[] | undefined;

### parameters

| Name                 | Type                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            | Description |
| -------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------- |
| `__namedParameters*` | **config**: ### properties| Name         | Type                                | Description                                                                                                                                    | | ------------ | ----------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------- | | `BuildProps` | [BuildProps](#buildprops)           |                                                                                                                                                | | `categories` | [DocType](#doctype)\[]              | page types that are considered as categories fields as well                                                                                    | | `files`      | string \| string\[]                 | alternative naming for docz compatibility                                                                                                      | | `ignore`     | string\[]                           | files to ignore. by default \['readme.md', 'changelog.md', 'code_of_conduct.md', 'contributing.md', 'license.md']                              | | `instrument` | any                                 | instrumentation configuration                                                                                                                  | | `pages`      | [PagesOnlyRoutes](#pagesonlyroutes) | base url path for API documentation pages. Default is "docs/"                                                                                  | | `siteMap`    | [SitemapConfig](#sitemapconfig)     | if false, disable automatic sitemap generation                                                                                                 | | `siteRoot`   | string                              | the site base url, by default the site starts at /                                                                                             | | `siteUrl`    | string                              | Deployed site url. Also used for auto generated sitemap.                                                                                       | | `stories`    | string \| string\[]                 | wild card search string for the stories internally using \`glob\` for the search: https&#x3A;//www.npmjs.com/package/glob example: "./stories/ |**configPath**: string |             |
| `returns`            | string\[] \| undefined                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          |             |

## fixGlob

_defined in [@component-controls/config/src/index.ts](https://github.com/ccontrols/component-controls/tree/master/core/config/src/index.ts#L98)_

**function** fixGlob(`golbExpr`\*: string): string;

### parameters

| Name        | Type   | Description |
| ----------- | ------ | ----------- |
| `golbExpr*` | string |             |
| `returns`   | string |             |

## getConfigurationArg

return the configuration folder from command-line parameters
command line accepts -c/ -config parameter for config path
the config file is assumed named main.js/main.ts

_defined in [@component-controls/config/src/index.ts](https://github.com/ccontrols/component-controls/tree/master/core/config/src/index.ts#L37)_

**function** getConfigurationArg(`args`\*: string\[]): string | undefined;

### parameters

| Name      | Type                | Description |
| --------- | ------------------- | ----------- |
| `args*`   | string\[]           |             |
| `returns` | string \| undefined |             |

## loadConfiguration

 given a base project folder and a configuration folder, returns the configuration file

_defined in [@component-controls/config/src/index.ts](https://github.com/ccontrols/component-controls/tree/master/core/config/src/index.ts#L58)_

**function** loadConfiguration(`baseFolder`\*: string, `configFolder`: string, `args`: string\[], `defaultConfigPath`: string): [ConfigurationResult](#configurationresult) | undefined;

### parameters

| Name                | Type                                                     | Description                                    |
| ------------------- | -------------------------------------------------------- | ---------------------------------------------- |
| `baseFolder*`       | string                                                   | project folder to start the search with        |
| `configFolder`      | string                                                   | folder where the configuration file is located |
| `args`              | string\[]                                                | optional arguments                             |
| `defaultConfigPath` | string                                                   |                                                |
| `returns`           | [ConfigurationResult](#configurationresult) \| undefined |                                                |

## mergeBuildConfiguration

merge a configuration passed through cli or tools, with the build configration from the config path

_defined in [@component-controls/config/src/index.ts](https://github.com/ccontrols/component-controls/tree/master/core/config/src/index.ts#L169)_

**function** mergeBuildConfiguration(`config`\*: [BuildConfiguration](#buildconfiguration)): [BuildConfiguration](#buildconfiguration);

### parameters

| Name      | Type                                      | Description |
| --------- | ----------------------------------------- | ----------- |
| `config*` | [BuildConfiguration](#buildconfiguration) |             |
| `returns` | [BuildConfiguration](#buildconfiguration) |             |

<!-- END-TSDOC-TYPESCRIPT -->
