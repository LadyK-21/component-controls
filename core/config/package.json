{
  "name": "@component-controls/config",
  "version": "3.11.0",
  "description": "Component controls configuration reading utilities",
  "main": "dist/index.js",
  "module": "dist/index.esm.js",
  "typings": "dist/index.d.ts",
  "files": [
    "dist/",
    "package.json",
    "README.md"
  ],
  "scripts": {
    "build": "yarn cross-env NODE_ENV=production rollup -c",
    "dev": "yarn cross-env NODE_ENV=development rollup -cw",
    "docs": "ts-md",
    "fix": "yarn lint --fix",
    "lint": "yarn eslint . --ext mdx,ts,tsx",
    "prepare": "yarn build",
    "test": "yarn jest"
  },
  "homepage": "https://github.com/ccontrols/component-controls",
  "bugs": {
    "url": "https://github.com/ccontrols/component-controls/issues"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/ccontrols/component-controls.git",
    "directory": "core/config"
  },
  "license": "MIT",
  "dependencies": {
    "@component-controls/core": "^3.11.0",
    "@component-controls/instrument": "^3.11.0",
    "@component-controls/render": "^3.11.0",
    "glob": "^7.1.6",
    "glob-base": "^0.3.0",
    "micromatch": "^4.0.2",
    "typescript": "^4.0.5",
    "yargs": "^15.3.1"
  },
  "devDependencies": {
    "@component-controls/ts-markdown-docs": "^3.8.2",
    "@types/faker": "^4.1.9",
    "@types/glob": "^7.1.3",
    "@types/glob-base": "^0.3.0",
    "@types/jest": "^26.0.10",
    "@types/micromatch": "^4.0.1",
    "cross-env": "^5.2.1",
    "eslint": "^6.5.1",
    "jest": "^26.4.2"
  },
  "esm": {
    "cjs": true
  },
  "browser": {
    "fs": false
  },
  "jest": {
    "testEnvironment": "node",
    "roots": [
      "."
    ]
  },
  "publishConfig": {
    "access": "public"
  },
  "authors": [
    "Atanas Stoyanov"
  ],
  "gitHead": "e30d62f101e104711a16261fce04785d58cac1eb"
}
