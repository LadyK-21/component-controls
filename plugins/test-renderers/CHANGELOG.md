# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [3.10.0](https://github.com/ccontrols/component-controls/compare/v3.9.1...v3.10.0) (2021-05-08)


### Features

* document-level data-driven tests ([806b16a](https://github.com/ccontrols/component-controls/commit/806b16a622e839ece9fca625e83a6d6427b49753))
* initial values data testing ([3f5d298](https://github.com/ccontrols/component-controls/commit/3f5d2983c1f589cd4dd297d231309fbcc3b9c7da))





# [3.9.0](https://github.com/ccontrols/component-controls/compare/v3.8.3...v3.9.0) (2021-04-30)

**Note:** Version bump only for package @component-controls/test-renderers





## [3.8.2](https://github.com/ccontrols/component-controls/compare/v3.8.1...v3.8.2) (2021-04-30)


### Bug Fixes

* issue when export function name missing ([57bb2e5](https://github.com/ccontrols/component-controls/commit/57bb2e5363f40f19f01d231a5bcc5d931d72e7c8))





# [3.8.0](https://github.com/ccontrols/component-controls/compare/v3.7.0...v3.8.0) (2021-04-29)


### Features

* add accessibility tests to templates ([f58f4ae](https://github.com/ccontrols/component-controls/commit/f58f4ae6e3f199e63bf1fc9198a2c7caa0942ed0))





# [3.7.0](https://github.com/ccontrols/component-controls/compare/v3.6.3...v3.7.0) (2021-04-20)


### Bug Fixes

* ts-jest memory leak ([71f714d](https://github.com/ccontrols/component-controls/commit/71f714d74e2cea3d9705c1149f8d7ab2196744fc))





## [3.6.3](https://github.com/ccontrols/component-controls/compare/v3.6.2...v3.6.3) (2021-04-08)

**Note:** Version bump only for package @component-controls/test-renderers





## [3.6.2](https://github.com/ccontrols/component-controls/compare/v3.6.1...v3.6.2) (2021-04-07)

**Note:** Version bump only for package @component-controls/test-renderers





## [3.6.1](https://github.com/ccontrols/component-controls/compare/v3.6.0...v3.6.1) (2021-04-06)

**Note:** Version bump only for package @component-controls/test-renderers





# [3.6.0](https://github.com/ccontrols/component-controls/compare/v3.5.5...v3.6.0) (2021-04-06)

**Note:** Version bump only for package @component-controls/test-renderers





## [3.5.3](https://github.com/ccontrols/component-controls/compare/v3.5.2...v3.5.3) (2021-04-04)

**Note:** Version bump only for package @component-controls/test-renderers





# [3.5.0](https://github.com/ccontrols/component-controls/compare/v3.4.5...v3.5.0) (2021-04-04)

**Note:** Version bump only for package @component-controls/test-renderers
