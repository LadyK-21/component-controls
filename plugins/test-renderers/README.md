# Table of contents

- [Overview](#overview)
- [Installation](#installation)
- [API](#api)
  - [<ins>reactRunDOM</ins>](#insreactrundomins)
  - [<ins>renderDocument</ins>](#insrenderdocumentins)
  - [<ins>renderExample</ins>](#insrenderexampleins)

# Overview

A collection of test renderers for react

# Installation

This package is usually installed as part of the @component-controls package, but you can also install it standalone:

```bash
$ npm install @component-controls/test-renderers --save-dev
```

# API

<react-docgen-typescript path="./src" />

<!-- START-REACT-DOCGEN-TYPESCRIPT -->

## <ins>reactRunDOM</ins>

run some function in the dom, by using react-dom

_reactRunDOM [source code](https://github.com/ccontrols/component-controls/tree/master/plugins/test-renderers/src/react-run-dom.ts)_

## <ins>renderDocument</ins>

returns an array of the rendered all stories(examples) imports from a document using a test framework.

_renderDocument [source code](https://github.com/ccontrols/component-controls/tree/master/plugins/test-renderers/src/render-document.ts)_

## <ins>renderExample</ins>

render an example using a test framework.

_renderExample [source code](https://github.com/ccontrols/component-controls/tree/master/plugins/test-renderers/src/render-example.ts)_

### properties

| Name       | Type                                                                                                                                                                                                                                                                           | Description |
| ---------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | ----------- |
| `example*` | _Example&lt;Props>_                                                                                                                                                                                                                                                            |             |
| `doc*`     | _{ \[key: string]: any; title: string; type?: string; route?: string; date?: Date; dateModified?: Date; draft?: boolean; tags?: string\[]; keywords?: string\[]; description?: string \| Element; ... 12 more ...; parameters?: any; } & StoryProps&lt;...> & PageLayoutProps_ |             |
| `config*`  | _string \| RunConfiguration_                                                                                                                                                                                                                                                   |             |
| `controls` | _ExampleControls_                                                                                                                                                                                                                                                              |             |

<!-- END-REACT-DOCGEN-TYPESCRIPT -->
