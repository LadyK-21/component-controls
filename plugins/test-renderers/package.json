{
  "name": "@component-controls/test-renderers",
  "version": "3.6.0",
  "description": "React test renderers: rtl, rtr and enzyme",
  "keywords": [
    "react",
    "react-testing-library",
    "react-test-renderer",
    "enzyme",
    "testing"
  ],
  "main": "dist/index.js",
  "module": "dist/index.esm.js",
  "typings": "dist/index.d.ts",
  "files": [
    "dist/",
    "package.json",
    "README.md"
  ],
  "scripts": {
    "build": "yarn cross-env NODE_ENV=production rollup -c",
    "dev": "yarn cross-env NODE_ENV=development rollup -cw",
    "docs": "ts-md",
    "fix": "yarn lint --fix",
    "lint": "yarn eslint . --ext mdx,ts,tsx",
    "prepare": "yarn build",
    "test": "yarn jest"
  },
  "homepage": "https://github.com/ccontrols/component-controls",
  "bugs": {
    "url": "https://github.com/ccontrols/component-controls/issues"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/ccontrols/component-controls.git",
    "directory": "plugins/test-renderers"
  },
  "license": "MIT",
  "dependencies": {
    "@component-controls/config": "^3.6.0",
    "@component-controls/core": "^3.6.0",
    "@component-controls/render": "^3.6.0",
    "jest": "^26.4.2",
    "path": "^0.12.7",
    "react-dom": "^16.13.1"
  },
  "devDependencies": {
    "typescript": "^4.0.5"
  },
  "publishConfig": {
    "access": "public"
  },
  "authors": [
    "Atanas Stoyanov"
  ],
  "jest": {
    "preset": "ts-jest",
    "roots": [
      "./test"
    ]
  },
  "gitHead": "e30d62f101e104711a16261fce04785d58cac1eb"
}
