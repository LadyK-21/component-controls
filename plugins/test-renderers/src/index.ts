export { renderExample } from './render-example';
export { renderDocument } from './render-document';
export { reactRunDOM } from './react-run-dom';

export const renderErr = (): void => {
  throw new Error('Could not render the story');
};

export const componentErr = (): void => {
  throw new Error('Error rendering component with react-test-renderer');
};
