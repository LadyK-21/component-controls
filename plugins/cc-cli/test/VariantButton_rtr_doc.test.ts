import * as path from 'path';
import { run, AxeResults } from 'axe-core';
import { reactRunDOM } from '@component-controls/test-renderers';
import '@component-controls/jest-axe-matcher';
import { loadConfigurations } from '@component-controls/config';
import { renderDocument } from '@component-controls/test-renderers';
import renderer, { act } from 'react-test-renderer';
import { renderErr, componentErr } from '@component-controls/test-renderers';

import * as examples from './fixtures/VariantButton.docs';
import data from '../../../core/jest-extract/test/fixtures/story/VariantButton.data';

describe('VariantButton', () => {
  const configPath = path.resolve(__dirname, '.config');
  const config = loadConfigurations(configPath);
  let renderedExamples: ReturnType<typeof renderDocument> = [];
  act(() => {
    renderedExamples = renderDocument(examples, config, data);
  });
  if (!renderedExamples.length) {
    renderErr();
    return;
  }
  renderedExamples.forEach(({ name, rendered, dataId, values }) => {
    describe(name, () => {
      const runTests = () => {
        it('snapshot', () => {
          const component = renderer.create(rendered);
          if (!component) {
            componentErr();
            return;
          }
          expect(component.toJSON()).toMatchSnapshot();
        });
        it('accessibility', async () => {
          const axeResults = await reactRunDOM<AxeResults>(rendered, run);
          expect(axeResults).toHaveNoAxeViolations();
        });
      };
      if (values) {
        describe(dataId, runTests);
      } else {
        runTests();
      }
    });
  });
});
