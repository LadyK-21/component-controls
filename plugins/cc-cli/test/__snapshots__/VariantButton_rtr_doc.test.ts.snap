// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VariantButton accent snapshot 1`] = `
<button
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#9B51E0",
      "border": "1px solid #000000",
      "borderRadius": 8,
      "color": "#F2F2F2",
      "display": "flex",
    }
  }
>
  <div
    data-testid="label"
    style={
      Object {
        "padding": "0 10px",
      }
    }
  >
    Accent
  </div>
</button>
`;

exports[`VariantButton disabled snapshot 1`] = `
<button
  disabled={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#E0E0E0",
      "border": "1px solid #000000",
      "borderRadius": 8,
      "color": "#828282",
      "display": "flex",
    }
  }
>
  <div
    data-testid="label"
    style={
      Object {
        "padding": "0 10px",
      }
    }
  >
    Disabled
  </div>
</button>
`;

exports[`VariantButton error snapshot 1`] = `
<button
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#EB5757",
      "border": "1px solid #000000",
      "borderRadius": 8,
      "color": "#F2F2F2",
      "display": "flex",
    }
  }
>
  <div
    data-testid="label"
    style={
      Object {
        "padding": "0 10px",
      }
    }
  >
    Error
  </div>
</button>
`;

exports[`VariantButton overview 0 snapshot 1`] = `
<button
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2F80ED",
      "border": "1px solid #000000",
      "borderRadius": 8,
      "color": "#F2F2F2",
      "display": "flex",
    }
  }
>
  <div
    data-testid="label"
    style={
      Object {
        "padding": "0 10px",
      }
    }
  >
    Gisselle Mohr
  </div>
</button>
`;

exports[`VariantButton overview 1 snapshot 1`] = `
<button
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2F80ED",
      "border": "1px solid #000000",
      "borderRadius": 8,
      "color": "#F2F2F2",
      "display": "flex",
    }
  }
>
  <div
    data-testid="label"
    style={
      Object {
        "padding": "0 10px",
      }
    }
  >
    Barbara Brakus
  </div>
</button>
`;

exports[`VariantButton overview 2 snapshot 1`] = `
<button
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2F80ED",
      "border": "1px solid #000000",
      "borderRadius": 8,
      "color": "#F2F2F2",
      "display": "flex",
    }
  }
>
  <div
    data-testid="label"
    style={
      Object {
        "padding": "0 10px",
      }
    }
  >
    Adah Nolan
  </div>
</button>
`;

exports[`VariantButton overview 3 snapshot 1`] = `
<button
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2F80ED",
      "border": "1px solid #000000",
      "borderRadius": 8,
      "color": "#F2F2F2",
      "display": "flex",
    }
  }
>
  <div
    data-testid="label"
    style={
      Object {
        "padding": "0 10px",
      }
    }
  >
    Jon Will
  </div>
</button>
`;

exports[`VariantButton overview 4 snapshot 1`] = `
<button
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2F80ED",
      "border": "1px solid #000000",
      "borderRadius": 8,
      "color": "#F2F2F2",
      "display": "flex",
    }
  }
>
  <div
    data-testid="label"
    style={
      Object {
        "padding": "0 10px",
      }
    }
  >
    Mr. Dulce Rice
  </div>
</button>
`;

exports[`VariantButton overview 5 snapshot 1`] = `
<button
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2F80ED",
      "border": "1px solid #000000",
      "borderRadius": 8,
      "color": "#F2F2F2",
      "display": "flex",
    }
  }
>
  <div
    data-testid="label"
    style={
      Object {
        "padding": "0 10px",
      }
    }
  >
    Mrs. Jermey Jacobson
  </div>
</button>
`;

exports[`VariantButton overview 6 snapshot 1`] = `
<button
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2F80ED",
      "border": "1px solid #000000",
      "borderRadius": 8,
      "color": "#F2F2F2",
      "display": "flex",
    }
  }
>
  <div
    data-testid="label"
    style={
      Object {
        "padding": "0 10px",
      }
    }
  >
    Clark Hickle
  </div>
</button>
`;

exports[`VariantButton overview 7 snapshot 1`] = `
<button
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2F80ED",
      "border": "1px solid #000000",
      "borderRadius": 8,
      "color": "#F2F2F2",
      "display": "flex",
    }
  }
>
  <div
    data-testid="label"
    style={
      Object {
        "padding": "0 10px",
      }
    }
  >
    Ibrahim Purdy
  </div>
</button>
`;

exports[`VariantButton overview 8 snapshot 1`] = `
<button
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2F80ED",
      "border": "1px solid #000000",
      "borderRadius": 8,
      "color": "#F2F2F2",
      "display": "flex",
    }
  }
>
  <div
    data-testid="label"
    style={
      Object {
        "padding": "0 10px",
      }
    }
  >
    Jody Legros
  </div>
</button>
`;

exports[`VariantButton overview 9 snapshot 1`] = `
<button
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2F80ED",
      "border": "1px solid #000000",
      "borderRadius": 8,
      "color": "#F2F2F2",
      "display": "flex",
    }
  }
>
  <div
    data-testid="label"
    style={
      Object {
        "padding": "0 10px",
      }
    }
  >
    Ford Kunze
  </div>
</button>
`;

exports[`VariantButton primary snapshot 1`] = `
<button
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2F80ED",
      "border": "1px solid #000000",
      "borderRadius": 8,
      "color": "#F2F2F2",
      "display": "flex",
    }
  }
>
  <div
    data-testid="label"
    style={
      Object {
        "padding": "0 10px",
      }
    }
  >
    Primary
  </div>
</button>
`;

exports[`VariantButton success snapshot 1`] = `
<button
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#219653",
      "border": "1px solid #000000",
      "borderRadius": 8,
      "color": "#F2F2F2",
      "display": "flex",
    }
  }
>
  <div
    data-testid="label"
    style={
      Object {
        "padding": "0 10px",
      }
    }
  >
    Success
  </div>
</button>
`;

exports[`VariantButton warning snapshot 1`] = `
<button
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#EB5757",
      "border": "1px solid #000000",
      "borderRadius": 8,
      "color": "#F2F2F2",
      "display": "flex",
    }
  }
>
  <div
    data-testid="label"
    style={
      Object {
        "padding": "0 10px",
      }
    }
  >
    Warning
  </div>
</button>
`;
